pkg/trtl/honu.go

```diff
@@ -30,72 +30,68 @@ func NewHonuService(s *Server) (*HonuService, error) {
 	var err error
 
 	if _, found := reservedNamespaces[in.Namespace]; found {
-		log.Warn().Msg("cannot use reserved namespace")
+		log.Warn().Str("namespace", in.Namespace).Msg("cannot use reserved namespace")
 		return nil, status.Error(codes.PermissionDenied, "cannot use reserved namespace")
 	}
 	if len(in.Key) == 0 {
 		log.Warn().Msg("missing key in Trtl Get request")
 		return nil, status.Error(codes.InvalidArgument, "key must be provided in Get request")
 	}
 
-	var key []byte
-	if len(in.Namespace) > 0 {
-		key = prepend(in.Namespace, in.Key)
-	} else {
-		key = prepend("default", in.Key)
-	}
-
-	if in.Options != nil {
-		log.Debug().Str("key", string(key)).Bool("return_meta", in.Options.ReturnMeta).Msg("Trtl Get")
-	} else {
-		log.Debug().Str("key", string(key)).Msg("Trtl Get")
-	}
 	if in.Options != nil && in.Options.ReturnMeta {
-		// Retrieve and return the metadata.
+		// Retrieve and return the metadata (uses honu.Object())
+		log.Debug().Str("key", string(in.Key)).Bool("return_meta", in.Options.ReturnMeta).Msg("Trtl Get")
+
+		// Check if we have a namespace
 		var object *object.Object
-		if object, err = h.db.Object(key); err != nil {
+		if in.Namespace != "" {
+			object, err = h.db.Object(in.Key, options.WithNamespace(in.Namespace))
+		} else {
+			object, err = h.db.Object(in.Key)
+		}
+
+		if err != nil {
 			// TODO: Check for the honu not found error instead.
 			if err == leveldb.ErrNotFound {
-				log.Debug().Err(err).Str("key", string(key)).Msg("specified key not found")
+				log.Debug().Err(err).Str("key", string(in.Key)).Msg("specified key not found")
 				return nil, status.Error(codes.NotFound, err.Error())
 			}
-			log.Error().Err(err).Str("key", string(key)).Msg("unable to retrieve object")
+			log.Error().Err(err).Str("key", string(in.Key)).Msg("unable to retrieve object")
 			return nil, status.Error(codes.Internal, err.Error())
 		}
 		return &pb.GetReply{
 			Value: object.Data,
-			Meta: &pb.Meta{
-				Key:       object.Key,
-				Namespace: object.Namespace,
-				Region:    object.Region,
-				Owner:     object.Owner,
-				Version: &pb.Version{
-					Pid:     object.Version.Pid,
-					Version: object.Version.Version,
-					Region:  object.Version.Region,
-				},
-				Parent: &pb.Version{
-					Pid:     object.Version.Parent.Pid,
-					Version: object.Version.Parent.Version,
-					Region:  object.Version.Parent.Region,
-				},
-			},
+			Meta:  returnMeta(*object),
 		}, nil
 	}
 
-	// Just return the value for the given key.
+	// No metadata requested; just return the value for the given key (uses honu.Get())
+	log.Debug().Str("key", string(in.Key)).Msg("Trtl Get")
+
+	// But we do have to check if we have a namespace
 	var value []byte
-	if value, err = h.db.Get(key); err != nil {
-		// TODO: Check for the honu not found error instead.
+	if in.Namespace != "" {
+		value, err = h.db.Get(in.Key, options.WithNamespace(in.Namespace))
+	} else {
+		value, err = h.db.Get(in.Key)
+	}
```

Comment:
We'll probably want to abstract this out as well at some point to maintain our own sanity.

Reply:
Great idea! Any suggestions?